.github/workflows/pkgci_regression_test_a100.yml

```diff
@@ -0,0 +1,51 @@
+# Copyright 2023 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+name: PkgCI Regression Test (A100)
+on:
+  workflow_call:
+    inputs:
+      artifact_run_id:
+        type: string
+        default: ""
+  workflow_dispatch:
+    inputs:
+      artifact_run_id:
+        type: string
+        default: ""
+
+jobs:
+  linux_x86_64:
+    name: Linux (x86_64)
+    runs-on: nodai-nvgpu-a100-x86-64
```

Comment:
How many runners are there with this configuration? If the runner(s) have trouble keeping up with demand, we could change to T4 machines or postsubmit only testing. We switched which machines ran at which stages in the ci.yml jobs following inventory shortages, context in https://github.com/openxla/iree/pull/14765 and https://github.com/openxla/iree/issues/14169.

You could also mention in the PR description that these use new (persistent?) self-hosted runners, not the cloud auto-scaling runners used in other workflows.

Reply:
Another thing that could help with usage would be connecting [`build_tools/github_actions/configure_ci.py`](https://github.com/openxla/iree/blob/main/build_tools/github_actions/configure_ci.py) to [`.github/workflows/pkgci.yml`](https://github.com/openxla/iree/blob/main/.github/workflows/pkgci.yml). That does some filtering so PRs that only change docs and other non-code files don't run all the builds/tests.